backend/src/main/java/org/sefglobal/core/academix/service/SubCategoryService.java

```diff
@@ -40,4 +41,20 @@ public SubCategoryService(SubCategoryRepository subCategoryRepository, CategoryR
         }
         return subCategoryRepository.getAllByCategory_Id(categoryId);
     }
+
+    /**
+     * Retrieves the subcategory with the requested id
+     * @param subCategoryId which is the id of the requested subCategory
+     * @return CustomSubCategory object
+     * @throws ResourceNotFoundException if a subcategory with the requested id doesn't exist
+     */
+    public CustomSubCategory getSubCategoryById(Long subCategoryId) throws ResourceNotFoundException {
+        Optional<CustomSubCategory> subCategory = subCategoryRepository.findBy_Id(subCategoryId);
+        if(!subCategory.isPresent()){
+            String msg = "Error, Sub Category by id:"+subCategoryId+" doesn't exist.";
```

Comment:
I think these lines are not formatted. 

format the code

Reply:
Ah yes. 
Resolved